### DISCUSSION TITLE
@expresbodhi's Pool

### DISCUSSION THREAD

**Discussion by expresbodhi** [tagyoureit/nodejs-poolController#788] [My Pool]
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Documenting my journey to Share Body dream system.<span class="Apple-converted-space">&nbsp; </span>I bookmarked this project last year and decided to take the plunge when my Intermatic timer started tripping the breaker. We purchased the house in 2021 and haven’t done much in the way of improving the pool to date. My goals are to get to the stand alone dream system and then eventually add a Chlorinator.<span class="Apple-converted-space">&nbsp;</span></p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 12.0px 'Helvetica Neue'; color: #000000"><b>Starting Configuration and Background</b><span style="font: 11.0px 'Helvetica Neue'"><span class="Apple-tab-span" style="white-space:pre">	</span></span></p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Type of Pool: Plaster</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Pool Volume: 23,000 Gallons</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Pump: Pentair IntelliFlo VSF Pool Pump 3.0 hp</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Filter: Pentair Clean and Clear Plus CCP420</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Chlorinators: None</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Chemistry Controllers: None</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Valves: Three manual<span class="Apple-converted-space">&nbsp;</span></p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Heaters: Pentair Mini Max 150</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Lights: Pool and Spa</p>
<p style="margin: 0.0px 0.0px 0.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">Sensors: None</p>
<p style="margin: 0.0px 0.0px 16.0px 0.0px; font: 12.0px 'Helvetica Neue'; color: #000000"><b>Shared Body Dream System (Pool/Spa)</b></p>
<p style="margin: 0.0px 0.0px 16.0px 0.0px; font: 10.0px 'Helvetica Neue'; color: #000000">Almost all costs include taxes and shipping.</p>

Item | Retailer | Cost | Notes
-- | -- | -- | --
Altelix 14x12x6 Vented Industrial DIN Rail Fiberglass Enclosure Fiberglass | Altelix | $199.61 | Includes additional Din mount $7.19. Shipping was $32.43!!!
Raspberry Pi 4 Model B - 8 GB RAM | Ada Fruit | $88.11 | Followed @rpilocator on Twitter for stock updates
Lexar 128GB Micro SD Card | Amazon | $14.00 |  
DIN-RAIL Kit Type 2 Perpendicular Mount for Raspberry Pi | Amazon | $10.78 | sequent microsystems mount
Smart Fan HAT | Sequent Microsystems | $25.00 | V4.0
Mean Well Model SDR-240-24 | Ebay | $79.99 |  
Mean Well MDR-60-24 24V 2.5A 60W | Amazon | $30.70 |  
Building Automation V4  8-Layer Stackable HAT | Sequent Microsystems | $95.00 |  
Eight Relays 4A/120V  8-Layer Stackable HAT | Sequent Microsystems | $45.00 |  
Temperature Sensors,  Air/Water/Solar 20' Cable For Pentair | Ebay | $37.79 | Pack of 2
Pentair Intelliflo/Intellipro 350122 VS VSF Comm Data Cable | Ebay | $36.26 |  
Wago TopJob Din rail connectors | Digi-Key | $66.31 | Connectors, Endplates, End Stops, etc.

<p style="margin: 0.0px 0.0px 16.0px 0.0px; font: 12.0px 'Helvetica Neue'; color: #000000"><b>Current Status:</b></p>
<p style="margin: 0.0px 0.0px 16.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">I have the everything to complete the dream system in my garage except the valves.<span class="Apple-converted-space">&nbsp;</span></p>
<p style="margin: 0.0px 0.0px 16.0px 0.0px; font: 11.0px 'Helvetica Neue'; color: #000000">My first real question is I see some users using the eaton 10a din rail circuits in their enclosures. Would those be suitable to use for the pool lights? I would like to move those switches into the enclosure. The current setup doesn’t have a gfci outlet so I would like to add one of those as well.<span class="Apple-converted-space">&nbsp;</span></p>

![IMG_1929](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/ddc9ff96-4694-4212-84cd-66861a3b0abb)
![IMG_1930](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/63c1aff0-b458-4010-af40-a640f8626da4)
![IMG_1932](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/978a0024-e464-49ef-ab03-b74914c3b9a0)

**Comment by rstrouse** [comment]
Welcome and good luck on the journey.  Btw I found your football.

**Comment by rstrouse** [comment]
On your questions... First, what kind of pool lights do you have?  Are these incandescent 120v or do they attach to a transformer?  Second, circuit breakers are used to protect the wiring not the device so what gage is the wiring from the lights to timer.  Also, is there a GFCI outlet in the box below the switches?

**Reply by expresbodhi** [reply]
I believe the they are 120v as there is no transformer and the wiring looks like all 12 gage was used. The Box below the light switches is a switch for the VSF Pool Pump.

**Reply by rstrouse** [reply]
It really is bad that you have 120v lights in a pool that do not have a GFCI.  So that being said you are looking for a UL489 GFCI breaker that is less than the rating of the conductor.  This means that it will trip before meltdown.  So this means that if the wire is 12awg you can use up to a 20A interrupter and if the wire is 14awg (please check to be sure) then you can use up to 15A.  If these are incandescent then they will draw a whole lot more power than if they are LED.  

Keep in mind I would not use a UL1077 device for this in any case because if you do convert to 120v LED lights there is still a transformer that is located in each fixture.  UL489 is for branch circuit protection and if they are incandescent then 10A is not the right breaker, you are looking for 15A on 14awg.  Or greater than 10A but less than 20A for 12awg wire.  If these are LED then 10A will do but Eaton also makes these in higher ratings.  Remember you are protecting the wire not the light.

**Reply by rock-crusher** [reply]
`my Intermatic timer started tripping the breaker`

Hi expresbodhi,

I would like to just mention that your Intermatic timer is not likely the culprit tripping the breaker. It’s either the line wires coming in, or the load wires going out to the equipment, or the equipment itself. It all depends when the breaker trips, when the time switch is off, or on. Now, it looks like your setup was originally for a single speed pump that has been changed out to the IntelliFlo VSF. That said, the load side wires in the time clock that went to the single speed pump have been relocated to the same terminals as the line side wire terminals coming in. This would be done so the IntelliFlo would receive direct unswitched power at all times. 

I’m going to assume the breaker is tripping when the time switch is off. Reason being, the heater seems to be the only thing connected to the load side of the timer switch now. If it only trips when the time switch is on, then something wrong with heater, or associated wiring.

You could try turning off the disconnect switch going to the VSF to eliminate it as being the problem.

If the breaker still trips after that with both switches in the off position (timer & VSF) it probably the line wires coming in have shorted. This could be anywhere in the conduit run from the beginning to the end. 

From your pictures, it could be the line wires coming through any of the boxes shown in the pictures as the boxes appear to be over crowded internally easily causing a short.

`It really is bad that you have 120v lights in a pool that do not have a GFCI`

As for the pool lights not having a GFCI, don’t be so sure about that. If I’m not mistaken, all permanently installed 120v underwater pool lighting since 1968 required GFCI according to the NEC. This does not mean all areas at the time adopted the ’68 code, but it was at that time included in article 680. I don’t know when the pool was built, but if it was sometime in the mid 1970s or later the pool light no doubt would have had GFCI protection in order to pass inspection. 

Just because we don’t see GFCI protection out at the pad location doesn’t mean the lights are not protected. Check the house electrical panel for a GFCI breaker, or a GFCI outlet anywhere along the conduit path between the house panel and the pool equipment.

Unfortunately, it is possible for someone over the years to have removed it not realizing how critical a role it plays in safety.

Also, what is the box facing backward in the picture; could this be the GFCI outlet?

<img width="600" alt="Unknown Box 2" src="https://github.com/tagyoureit/nodejs-poolController/assets/94593954/d6de6ee3-eec1-4389-8114-a340e23fac13">

**Reply by expresbodhi** [reply]
The Box Facing backwards is the Junction box for the lights. From what I know, there used to be a pump house that the previous owner took out. Maybe the GFCI was removed then? I looked at the inspection documents and saw that the GFCI for the pool was not checked. I followed the conduit from the panel to the pool and there is no GFCI in line.  

Safest thing for me at this point seems like I should put in a load center with a GFCI.

**Reply by expresbodhi** [reply]
I found a photo when we moved in and there was an additional junction box in line, so maybe that contained the GFCI? A acquaintance installed the enclosure and shortened the conduit a foot or so maybe responsible for removing it. Oddly, I had a 15A breaker turned off at the panel, which happened to be the light circuit. 

Since the conduit run is too small for a 60A load center, I will wait until we remodel the yard in the coming years. So I simply added a GFCI for the light circuit. In doing so, the pump breaker tripping issue seems to have alleviated itself. Was able to run the heater and pump with no issues. 

![IMG_1958](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/29c50e86-9c34-4c9d-9b56-1e87180a28fd)

**Reply by rstrouse** [reply]
At least you have a GFCI in there now.  What size conduit is run and what type is it.  Is it EMT?

**Comment by expresbodhi** [comment]
![IMG_2149](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/407dd22e-0b23-4f19-8984-a55d4db5cbb9)
 Quick update, Finally was able to get my hands on a few more items, Including two AquaLink RS JVA4424 Valve Actuators ($188.30 each via ebay, photo only has one installed on the ElectroCookie) and fired it up.  The current status is two valves, RS485 to my Pentair pump and both Air and Water temperatures. As you can see I have some work to do with wire clean up. Next steps will be to wire the heater. 

Day 1 I ran without the Vales using the schedule function, and now I am working to integrate the Valves and trying to understand how the circuits work. Getting there.

**Reply by expresbodhi** [reply]
Is the Valve behavior such that In Spa Mode Both Valves Are On, but while in Pool mode only the suction side is active? Or do I have my switches on my valves incorrect?  My thought was the "aways on" position for the Return would be the Spa, While the "always On" position for the suction to be from the Pool, so I wouldn't drain the Spa if something happened to the pi. 

Spa Mode:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/e9f8cc27-9d96-43bd-b2b6-9e2a376869b0)
![image](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/ad24ada8-3cbe-4a54-93c4-50b2adfdc687)
Pool Mode:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/5fe4b48c-6eb0-4bfa-be02-c32362b72d4e)
![image](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/f7e24bf2-cfd9-46a2-8996-f9950389f58d)

**Reply by rock-crusher** [reply]
Hi expresbodhi,
Great to see you’re moving forward with the project.
As for your latest questions, this link below will explain everything needed for your set-up:  
[Nixie Controller Setup](https://github.com/tagyoureit/nodejs-poolController/wiki/Nixie-Controller-Setup)
Just curious, looking at your controller and the equipment you have why the Mean Well 24vdc power supply? I don’t see a real purpose for it with your current set-up.  Is it something for future use?

**Reply by expresbodhi** [reply]
I guess I am a little confused because the Spillway feature and the Spa are behaving the same. Both Valves are active. Pool mode seems to work fine. Its just in spa mode, suction is coming from the pool.

**Reply by rock-crusher** [reply]
expresbodhi, 
I don’t think your set-up is correct.
Go back and make sure you have the Nixie Shared Body section set-up correctly:
•	The bodies tab
•	The filters tab
•	The valves tab
The link I posted ([Nixie Controller Setup](https://github.com/tagyoureit/nodejs-poolController/wiki/Nixie-Controller-Setup)) explains this in detail and if set-up correctly "Nixie will automatically control the position of these valves based upon the body that is currently turned on" i.e. pool/spa/spillway.
If you have this correct then something may be wrong with your wiring.

**Reply by expresbodhi** [reply]
It appears to be a configuration issue from the looks of the logs. Notice how the Spa Is changing both valves:
[11/08/2023, 20:19:02] verbose: Turning on a body circuit Spa
[11/08/2023, 20:19:02] verbose: Turning off shared body Pool circuit
[11/08/2023, 20:19:02] info: Clearing startup delays for Pool
[11/08/2023, 20:19:02] info: NCP: Setting Circuit Pool to false
[11/08/2023, 20:19:02] info: NCP: Setting Circuit Spa to true
[11/08/2023, 20:19:04] verbose: Nixie: Set valve 2-Return to true
[11/08/2023, 20:19:04] verbose: Nixie: Set valve 52-Intake to true

I re-imaged the Pi, did a new install and seems to be in the same state as before.

**Reply by rstrouse** [reply]
In spa mode both valves should be diverted (on).  In spillway mode only the return valve should be diverted (on).  And in pool mode neither valve should be diverted.   The default position for the valve should always be both undiverted (off).  If this is not operating your plumbing correctly you have one or more of the physical valves reversed via the valve switch.

**Reply by expresbodhi** [reply]
That was the piece that was missing. I thought when setting the default valves position it was based on what the end user wanted and that setting the connection in the aux-circuit would toggle the valve. Now I see the default really should be the pool body, and setting the connection under the Pool or Spa doesn't matter and can be left blank in my situation.  Thanks. 

Next Two Steps will be adding a pressure transducer to the filter and wiring the heater.

**Reply by expresbodhi** [reply]
Looks like my heater doesn't have a fireman's Switch.  It has a 3 wire connector for Automation, is there a suggested way to wire this without using the 3 wire,  as I understand the 3-wire connection is only for two fixed temperature set-points.
![123FA5C3-EEFC-4537-BD03-92A741EC2A43_1_105_c](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/d3f6f64a-3a02-4a0c-bc4a-71c64731e048)
![6CF8B7CD-8951-44DE-8D36-66914792F2DA_1_105_c](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/dadd770d-dea4-43a1-8829-0bbbb5aaa40d)

**Reply by rock-crusher** [reply]
![switch](https://github.com/tagyoureit/nodejs-poolController/assets/94593954/6735b9c6-fe19-4fad-8425-779e9cc929de)

**Comment by rstrouse** [comment]
There should be a jumper between two of the contacts.  Dig up the manual for it.

**Reply by rstrouse** [reply]
I don't know who makes this heater but the difference between a 3 wire and a 2 wire connection is often a jumper block between the pool/spa connector.  This is true for both Pentair and Hayward.  You should scare up the manual for this heater.

**Reply by expresbodhi** [reply]
The heater is branded Purex Triton Minimax 150, I don't know the history on when this was switched over to be under the Pentair brand. I found a copy of the manual [here](http://calypsoltd.com/wp-content/uploads/2013/01/Purex-Minimax-manual.pdf). 
![Screen Shot 2023-08-26 at 1 15 03 PM](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/b0662cb8-f920-4dfc-94ac-ca44f1c6a824)

**Reply by rock-crusher** [reply]
This heater was probably manufactured prior to the fancy automation systems we now have and the reason there is no fireman switch jumper loop.
In the diagram below, one way you can achieve the same result as the fireman loop is by opening and closing the circuit conductor I have circled via a relay on your hat.

![Fireman's Loop](https://github.com/tagyoureit/nodejs-poolController/assets/94593954/5e0fe039-584d-442f-b2a0-2f1390923236)

**Reply by rock-crusher** [reply]
For that matter you could just turn the power coming into the heater (120v) on and off via the hat relay as well.

**Reply by expresbodhi** [reply]
Sorry for all the questions on this, trying to reconcile the differences between the manual and the lid, Is this com wire the place you are referring to? The ignition control board is a honeywell s8600f.
![261815653-dadd770d-dea4-43a1-8829-0bbbb5aaa40d](https://github.com/tagyoureit/nodejs-poolController/assets/133036749/c9f54d00-eeee-4c7a-b4d3-77692ed8ac10)

**Reply by rock-crusher** [reply]
Best to go by the diagram on the heater cover.
 I think the easiest and best way for you to achieve this would be to do as indicated below by opening and closing the 24v circuit (this would be the same as toggling the on-off switch on the heater itself).
Once you do that, leave the heater switch in the on position and the thermostat set where you like it and the hat relay will now turn it on and off.

![Heater by pass switch](https://github.com/tagyoureit/nodejs-poolController/assets/94593954/ae213db0-4850-460c-ad30-f0db36576a7d)

**Reply by rock-crusher** [reply]
Edit… Since you have the water temp probe installed go ahead and turn the thermostat on the heater control panel to high and control the temp via Nixie.

**Reply by expresbodhi** [reply]
That was it, excellent! 
<img width="371" alt="Screen Shot 2023-08-27 at 8 10 12 PM" src="https://github.com/tagyoureit/nodejs-poolController/assets/133036749/8723370d-77c8-44cb-94ce-6fcbf985b73d">

**Reply by rock-crusher** [reply]
Awesome, happy to see you got it working; glad I could help you out  👍
